Library/Homebrew/download_strategy.rb

```diff
@@ -388,6 +388,25 @@ def _fetch
   end
 end
 
+class CurlSourceForgeSecureMirrorDownloadStrategy < CurlDownloadStrategy
+  def fetch
+    if !ENV["HOMEBREW_NO_INSECURE_REDIRECT"].nil?
+      oldurl = @url
+      @url = https_mirror
+      ohai "Sourceforge HTTPS: #{oldurl}"
+      ohai "=> converted to #{@url}"
+    end
+    super
+  end
+
+  def https_mirror
+     re = /(^https?:\/\/)(.+\.net\/project\/)(.+)/
+     @url.gsub(re) do
+       $1 + "www.mirrorservice.org/sites/download.sourceforge.net/pub/sourceforge/" + $3[0] + "/" + $3[0,2] + "/" + $3
```

Comment:
Use string interpolation instead of concatenation, e.g. `"#{$1}www...#{$3[0]}..."` instead of `$1 + "www..." + $3[0] + "..."`.